x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_fetch_custom_dashboards.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import type { InventoryItemType } from '@kbn/metrics-data-access-plugin/common';
+import type { InfraSavedCustomDashboard } from '../../../../common/custom_dashboards';
+import { InfraGetCustomDashboardsResponseBodyRT } from '../../../../common/http_api/custom_dashboards_api';
+import { useHTTPRequest } from '../../../hooks/use_http_request';
+import { throwErrors, createPlainError } from '../../../../common/runtime_types';
+import { useRequestObservable } from './use_request_observable';
+
+interface UseDashboardProps {
+  assetType: InventoryItemType;
+}
+
+export function useFetchCustomDashboards({ assetType }: UseDashboardProps) {
+  const { request$ } = useRequestObservable();
+
+  const decodeResponse = (response: any) => {
+    return pipe(
+      InfraGetCustomDashboardsResponseBodyRT.decode(response),
+      fold(throwErrors(createPlainError), identity)
+    );
+  };
+
+  const { error, loading, response, makeRequest } = useHTTPRequest<InfraSavedCustomDashboard[]>(
```

Comment:
Question: Why are we using `useHTTPRequest` here and in the `use_custom_dashboards.ts` we're using `useTrackedPromise`?

Reply:
I am trying to remember now - I think it was something around the option to pass parameters directly to the function ( as with `useHTTPRequest` there is no way to do it but in the case of GET endpoint we directly pass the asset type to the hook (a parameter we have from the props), In case of PUT/POST... we need to pass the params and it happens on Click so we have the params we hate to pass [here](https://github.com/jennypavlova/kibana/blob/176070-infra-implement-dashboard-tab-ui-in-host-details/x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/actions/save_dashboard_modal.tsx#L103-L110). With `createPromise` we can have a [function with parameters](https://github.com/jennypavlova/kibana/blob/176070-infra-implement-dashboard-tab-ui-in-host-details/x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_custom_dashboards.ts/#L181-L185) we can pass and do the request there 
I will see if I can use `useHTTPRequest` instead but I am not sure that it will work for those cases.

Reply:
Yeah. let's think about improving how we do HTTP requests in a separate ticket, and try to think about how we can make a better abstraction, similar to what APM does.